### ISSUE TITLE
Difficult to track changes

### ISSUE THREAD

**Issue by francoisvn** [adireddy/haxe-checkstyle#383]
Great tool, very useful for projects! However, when new version come out, it can be very difficult to track what has changed. New features can easily go missed. Is there a good way to track changes between versions and/or could you add a changelog that gets updated?

**Comment by AlexHaxe** [comment]
Hi, you are corect we don't have a CHANGELOG or an easy way to keep a detailed list of changes. 
When I made the latest release (which was the first one for me), I was making a mental note to figure out a way to keep a better track of changes for future releases.

In the releases section of github we usually list new features, but not all changes make the list.

As for the 2.2.0 release it's mainly a number of bugfixes in token tree parser (many were reported by @chipshort ), which should improve detection of issues for checks that rely on token tree. 

The only new feature and reason for jumping from 2.1.x to 2.2.0 is multithreading. Parsing and checkers now run in separate threads. Parsing means reading files from disc and building AST, token tree and line structures to be used in checks. Parsing is usually faster than checking, so there is only one fixed parser thread and n checker threads (default: 5). On Linux I saw a speed boost of up to 50%, on Windows it wasn't quite so high.
If you would like to meassure speed for yourself, there is a `-nothreads` command line switch that should   skip threads entirely and checkstyle should like in previous versions.
